src/authenticators/mod.rs

```diff
@@ -149,6 +160,14 @@ impl Admin {
 struct AdminList(Vec<Admin>);
 
 impl AdminList {
+    #[cfg_attr(
+        not(any(
+            feature = "direct-authenticator",
+            feature = "unix-peer-credentials-authenticator",
+            feature = "jwt-svid-authenticator",
+        )),
+        allow(unused)
+    )]
```

Comment:
I'm curious - why do them like this, adding `allow(unused)`, instead of `cfg(any(...))`?

Reply:
Because otherwise, if you don't compile in any authenticators, it warns (and thus fails to compile) with a message that doesn't scream "You forgot to enable an authenticator" to me.
So this is basically making sure that it compiles and then just doesn't ever work on runtime, instead of the case where it fails to compile with "weird" compile errors.
(Which one to choose was something I was doubting over for a while... I'm still convinced that I'd want something like the C `#error` to make the compilation fail with a self-set error.)

Reply:
And just after I mentioned that, I searched for the correct things and found https://doc.rust-lang.org/std/macro.compile_error.html